### PR TITLE
Fixes for publish package failure

### PR DESCRIPTION
**Pull request by pboyer**
### Purpose

This fixes two small regressions when publishing a new package.  The `PackageManagerClient` was not receiving the `packagesDirectory` argument. Also, the `RealFileInfo` class should return the `FileInfo.FullName` field, rather than `FileInfo.Name`.  

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers

@ikeough

### CODE REVIEW

**Comment by pboyer** [comment]
@ikeough ping - this is a 0.8.1 blocker.

**Comment by jnealb** [comment]
@ikeough @pboyer @mjk The code changes appear minor; let's push this across the finish line: master and RC8.1.

**Comment by mjk** [comment]
I think I was included accidentally, although it's very flattering.

File: test/DynamoCoreTests/PackageManager/PackageManagerClientTests.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void ListAll_ReturnsNonEmptyListWhenPackagesAreAvailable()
                 x.ExecuteAndDeserializeWithContent<List<PackageHeader>>(It.IsAny<HeaderCollectionDownload>()))
                 .Returns(mpl);
 
-            var m = new PackageManagerClient(c.Object, MockMaker.Empty<IPackageUploadBuilder>());
+            var m = new PackageManagerClient(c.Object, MockMaker.Empty<IPackageUploadBuilder>(), "");
```
Comment (ikeough) [inline test/DynamoCoreTests/PackageManager/PackageManagerClientTests.cs]: We could have a style discussion here about "" vs. `string.Empty`, but the internet seems to be split on the subject.

**Comment by ikeough** [comment]
LGTM.

**Pull request merged**